### PR TITLE
[FSTORE-501] Can't insert streaming dataframe in streaming feature group using `insert_stream` method

### PR DESCRIPTION
**Pull request by davitbzh**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [x] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/engine/spark.py [hunk 1]
```diff
@@ -214,12 +214,14 @@ def convert_to_default_dataframe(self, dataframe):
...
                 [col(x).alias(x.lower()) for x in dataframe.columns]
             )
-            nullable_schema = copy.deepcopy(lowercase_dataframe.schema)
-            for struct_field in nullable_schema:
-                struct_field.nullable = True
-            lowercase_dataframe = self._spark_session.createDataFrame(
-                lowercase_dataframe.rdd, nullable_schema
-            )
+            # for streaming dataframes this will be handled in DeltaStreamerTransformer.java class
+            if not lowercase_dataframe.isStreaming:
```
Comment (tdoehmen) [inline python/hsfs/engine/spark.py:218]: We might need the same change in the java engine here https://github.com/logicalclocks/feature-store-api/blob/master/java/src/main/java/com/logicalclocks/hsfs/engine/SparkEngine.java#L795

File: java/src/main/java/com/logicalclocks/hsfs/engine/hudi/DeltaStreamerTransformer.java [hunk 2]
```diff
@@ -30,11 +26,6 @@
 import org.apache.spark.sql.SparkSession;
 
 public class DeltaStreamerTransformer implements Transformer {
-  private FeatureStore featureStore;
```
Comment (moritzmeister) [inline java/src/main/java/com/logicalclocks/hsfs/engine/hudi/DeltaStreamerTransformer.java:33]: What happened here?

File: python/tests/core/test_feature_group_engine.py [hunk 3]
```diff
@@ -870,7 +870,8 @@ def test_insert_stream_online_enabled(self, mocker):
         )
         assert (
             mock_warnings_warn.call_args[0][0]
-            == "`insert_stream` method in the next release be available only for feature groups created with `stream=True`."
+            == "`insert_stream` method in the next release will be available only for feature groups created with "
+            "`stream=True`."
```
Comment (moritzmeister) [inline python/tests/core/test_feature_group_engine.py:874]: Does this work?

File: java/src/main/java/com/logicalclocks/hsfs/engine/hudi/DeltaStreamerTransformer.java (hunk 2 above)
Comment (davitbzh) [reply java/src/main/java/com/logicalclocks/hsfs/engine/hudi/DeltaStreamerTransformer.java:33]: Some of the code was left over and was not used at all. So I removed them

File: python/tests/core/test_feature_group_engine.py (hunk 3 above)
Comment (davitbzh) [reply python/tests/core/test_feature_group_engine.py:874]: seems to work

**Review by moritzmeister — APPROVED** [review]

**Pull request merged**